### PR TITLE
Env selector

### LINKED ISSUE #4531: [DASHBOARD] 1.7 Revisit environment selection dropdown
Revisit the way the environment selection, since this is not an action the user performs often. The dropdown could be replaced by something nicer.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #4531 #4595 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
Screenshot 2023-02-08 at 16 47 18

### CODE REVIEW

File: src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx [hunk 1]
```diff
@@ -44,13 +44,9 @@ test("GIVEN EnvironmentSelector and a project WHEN user clicks on toggle THEN li
     </MemoryRouter>
   );
 
-  const toggle = screen.getByRole("button", {
-    name: `Selected Project: ${envA.name} (${envA.projectName})`,
-  });
+  const toggle = screen.getByText(`Env: ${envA.name}`);
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx:47]: Environment: ...

File: src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx [hunk 2]
```diff
@@ -73,97 +69,19 @@ test("GIVEN EnvironmentSelector and populated store WHEN user clicks on an item
...
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx:72]: same here

File: src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx [hunk 3]
```diff
@@ -73,97 +69,19 @@ test("GIVEN EnvironmentSelector and populated store WHEN user clicks on an item
...
   expect(listItem).toBeVisible();
 
   await userEvent.click(listItem);
 
-  expect(
-    screen.queryByRole("button", {
-      name: `Selected Project: ${envB.name} (${envB.projectName})`,
-    })
-  ).toBeVisible();
+  expect(screen.getByText(`Env: ${envB.name}`)).toBeVisible();
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx:81]: here too

File: src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx [hunk 4]
```diff
@@ -182,7 +100,7 @@ test("GIVEN EnvironmentSelector and environments with identical names WHEN user
     </MemoryRouter>
   );
   const toggle = screen.getByRole("button", {
-    name: `Selected Project: ${envA.name} (${envA.projectName})`,
+    name: `Env: ${envB.name}`,
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx:103]: same

File: src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx [hunk 5]
```diff
@@ -191,7 +109,7 @@ test("GIVEN EnvironmentSelector and environments with identical names WHEN user
 
   expect(
     screen.getByRole("button", {
-      name: `Selected Project: ${envB.name} (${envB.projectName})`,
+      name: `Env: ${envB.name}`,
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx:112]: same

**Review by LauraStordeur — COMMENTED** [review]
could you add screenshots to the PR ? :D

File: src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx [hunk 6]
```diff
@@ -1,68 +1,93 @@
...
   isOpen,
   setIsOpen,
   toggleText,
 }) => {
   const { routeManager } = useContext(DependencyContext);
+  const envs = [
+    <DropdownGroup label={words("home.environment.selector")} key="envs-group">
+      {items.map((item, index) => (
+        <StyledItem onClick={() => onSelect(item)} key={`env-${index}-${item}`}>
+          {item.split(" ")[0].length > 22
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx:37]: That item has env name and project name, in parenthesis which is way to long, so decided to split it

[the remark this answers is not in the document]
Comment (LauraStordeur) [reply src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx:37]: Maybe confirm this ? Seems that project can be important in some situations :)

[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx:37]: confirmed, I can try this one

File: src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx [hunk 7]
```diff
@@ -1,68 +1,93 @@
...
-                  <Button variant="primary">
-                    {words("home.navigation.button")}{" "}
-                  </Button>
-                </Link>
-              </Tooltip>
-            </FlexItem>
-          </Flex>
-        </ContextSelectorFooter>
+      toggle={
+        <StyledToggle id="text" onToggle={() => setIsOpen(!isOpen)}>
```
Comment (LauraStordeur) [inline src/UI/Root/Components/Header/EnvSelector/EnvSelector.tsx:61]: id text? No better Id name ? :D

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**